python/ray/util/sgd/data/imagenet_dataset.py

```diff
@@ -0,0 +1,95 @@
+from ray.util.sgd.data.local_dataset import LocalDataset
+from PIL import Image
+import io
+import os
+import sys
+
+
+def pil_loader(raw_bytes):
+    with io.BytesIO(raw_bytes) as f:
+        img = Image.open(f)
+        return img.convert("RGB")
+
+
+# From torch.utils.data.ImageNet
+def _find_classes(dir):
+    """Finds the class folders in a dataset.
+
+    Args:
+        dir (string): Root directory path.
+
+    Returns:
+        tuple: (classes, class_to_idx) where classes are relative to
+    (dir), and class_to_idx is a dictionary.
+
+    Ensures:
+        No class is a subdirectory of another.
+
+    """
+    if sys.version_info >= (3, 5):
+        # Faster and available in Python 3.5 and above
+        classes = [d.name for d in os.scandir(dir) if d.is_dir()]
+    else:
+        classes = [
+            d for d in os.listdir(dir) if os.path.isdir(os.path.join(dir, d))
+        ]
+    classes.sort()
+    class_to_idx = {classes[i]: i for i in range(len(classes))}
+    return classes, class_to_idx
+
+
+def _load_val_sol(val_sol_path):
+    assert val_sol_path is not None, \
```

Comment:
Pretty sure Python will error if arguments without defaults are not provided anyway, so this will never trigger.

Reply:
I think this is just checking to make sure `None` isn't passed in as the argument.